ote_sdk/ote_sdk/entities/model_template.py

```diff
@@ -75,27 +75,138 @@ def __str__(self) -> str:
         return str(self.name)
 
 
+class TaskInfo(NamedTuple):
+    """
+    NamedTuple to store information about the task type like label domain, if it is
+    trainable, if it is an anomaly task and if it supports global or local labels.
+    """
+
+    domain: Domain
+    is_trainable: bool
+    is_anomaly: bool
+    is_global: bool
+    is_local: bool
+
+
 class TaskType(Enum):
     """
-    The type of algorithm within the task family.
+    The type of algorithm within the task family. Also contains relevant information
+    about the task type like label domain, if it is trainable, if it is an anomaly task
+    or if it supports global or local labels.
     """
 
-    NULL = auto()
-    DATASET = auto()
-    CLASSIFICATION = auto()
-    SEGMENTATION = auto()
-    DETECTION = auto()
-    ANOMALY_DETECTION = auto()
-    CROP = auto()
-    TILE = auto()
-    INSTANCE_SEGMENTATION = auto()
-    ACTIVELEARNING = auto()
-    ANOMALY_SEGMENTATION = auto()
-    ANOMALY_CLASSIFICATION = auto()
-    ROTATED_DETECTION = auto()
+    def __init__(
```

Comment:
https://docs.python.org/3/library/enum.html#when-to-use-new-vs-init

Reply:
Ah, thanks. Fixed.